cli/bazelisk/bazelisk.go

```diff
@@ -87,17 +89,39 @@ func ConfigureRunScript(args []string) (newArgs []string, scriptPath string, err
 }
 
 func InvokeRunScript(path string) (exitCode int, err error) {
+	done := make(chan struct{})
+	defer close(done)
+
 	if err := os.Chmod(path, 0o755); err != nil {
 		return -1, err
 	}
-	// TODO: Exec() replaces the current process, so it prevents us from running
-	// post-run hooks (if we decide those will be supported). If we want to use
-	// exec.Command() here instead of exec(), then we might need to manually
-	// forward signals from the parent file watcher process.
-	if err := syscall.Exec(path, nil, os.Environ()); err != nil {
+
+	cmd := exec.Command(path)
+	cmd.Stdout = os.Stdout
+	cmd.Stderr = os.Stderr
+	cmd.SysProcAttr = &syscall.SysProcAttr{Setpgid: true}
+	if err := cmd.Start(); err != nil {
+		return -1, err
+	}
+
+	// Force kill the child process if the CLI receives an interrupt.
+	sigs := make(chan os.Signal, 1)
+	signal.Notify(sigs, syscall.SIGINT, syscall.SIGTERM)
+	go func() {
+		select {
+		case <-sigs:
+			syscall.Kill(-cmd.Process.Pid, syscall.SIGKILL)
+		case <-done:
+		}
+	}()
```

Comment:
I think this will break some things, e.g. when pressing Ctrl+C when running the app with `bb run`, it should start a graceful shutdown, but I think this will force-kill the process?

I think it's surprising that we're not handling Ctrl+C properly since the current code is handing off control to the child process using `Exec()`, replacing the current process with the child. And I believe the generated `--run_script` should be doing the same thing - `exec()`-ing the child process so that the child process replaces it. So I'd think all signals should be getting directly handled by the binary that is run.

Maybe there's some unexpected process in the process tree that's intercepting signals? (Might be easy to check with a tool like `pstree`)